tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_type_runtime.bal

```diff
@@ -133,6 +133,20 @@ function testNeverWithAnyAndAnydataRuntime() {
     assertEquality(true, c4);
 }
 
+function testNeverFieldTypeCheck() {
```

Comment:
Let's also add some runtime tests that cover isLike checks.

Due to https://github.com/ballerina-platform/ballerina-lang/issues/33561 you may have to do something like

```ballerina
record { never i?; int j; } x = {j: 1};
anydata y = (<anydata> x).cloneReadOnly();
```

to get the immutable value at runtime.